### PR TITLE
Clarify auto-suggestion documentation

### PR DESCRIPTION
**Pull request by Colebow**
## Description

Cleanup on https://github.com/trinodb/trino/pull/11671 to make sure docs are consistent & clean.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Docs

## Related issues, pull requests, and links

* Relates to #11671 

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by Colebow** [comment]
image

File: docs/src/main/sphinx/installation/cli.rst [hunk 1]
```diff
@@ -431,12 +431,13 @@ mode:
 .. _cli-output-format:
 
 Auto suggestion
```
Comment (nineinchnick) [inline docs/src/main/sphinx/installation/cli.rst:433]: This should be moved outside of the `batch mode` section. Also, the anchor above refers to the `output formats` section. Looks like a result of a bad rebase.

Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst:433]: the anchor was messed up in the original documentation.. but yes.. we have to fix this

File: docs/src/main/sphinx/installation/cli.rst [hunk 2]
```diff
@@ -431,12 +431,13 @@ mode:
...
----------------
-As you type commands in the CLI , you see offered a suggestion after the cursor
-based on history of commands already entered.
+^^^^^^^^^^^^^^^
+
+The CLI will generate autocomplete suggestions based on command history.
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:436]: do not use future tense

**Review by mosabua — CHANGES_REQUESTED** [review]
Needs move of section

File: docs/src/main/sphinx/installation/cli.rst (hunk 1 above)
Comment (Colebow) [reply docs/src/main/sphinx/installation/cli.rst:433]: You think it makes sense to move this to the 'Running the CLI' section?

Comment (nineinchnick) [reply docs/src/main/sphinx/installation/cli.rst:433]: I'd move it after `history`, just before `batch mode`.

Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst:433]: that works

File: docs/src/main/sphinx/installation/cli.rst [hunk 3]
```diff
@@ -396,6 +396,15 @@ press :kbd:`Enter`.
 By default, you can locate the Trino history file in ``~/.trino_history``.
 Use the ``TRINO_HISTORY_FILE`` environment variable to change the default.
 
+Auto suggestion
+^^^^^^^^^^^^^^^
+
+The CLI generates autocomplete suggestions based on command history.
+
+Press :kbd:`→` to accept the suggestion and replace the current command line
+buffer. Press :kbd:`Option+→` to accept only the next keyword. Continue typing
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:405]: Option is only on a Mac keyboard ... what about windows and linux machines? Is it ALT ?  Or Windows key?

**Review by mosabua — CHANGES_REQUESTED** [review]
Needs one clarification for non macos users.. (and in fact the macos one should be secondary)

File: docs/src/main/sphinx/installation/cli.rst (hunk 3 above)
Comment (Colebow) [reply docs/src/main/sphinx/installation/cli.rst:405]: @nineinchnick do you know the answer to this?

Comment (ebyhr) [reply docs/src/main/sphinx/installation/cli.rst:405]: It will be `Ctrl+→` in Windows.

File: docs/src/main/sphinx/installation/cli.rst [hunk 4]
```diff
@@ -143,6 +143,8 @@ mode:
     - Enables display of debug information during CLI usage for
       :ref:`cli-troubleshooting`. Displays more information about query
       processing statistics.
+  * - ``--disable-autosuggestion``
```
Comment (ebyhr) [inline docs/src/main/sphinx/installation/cli.rst:146]: Please change to `--disable-auto-suggestion`. The original parameter was wrong.

**Review by ebyhr — APPROVED** [review]

File: docs/src/main/sphinx/installation/cli.rst (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst:146]: please fix asap @Colebow so we can get the PR merged

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**